src/charm.py

```diff
@@ -528,19 +521,19 @@ def _configure(self):  # noqa: C901
             v12_migration_date=v12_migration_date,
         ).build()
 
-        restart = self._update_config(config) or restart
+        restart = self._update_config(config)
+        new_layer = self._loki_pebble_layer
         self._loki_container.add_layer(self._name, new_layer, combine=True)
-        # Now that we for sure have a layer, we can check if the service is running
-        restart = not self._loki_container.get_service(self._service_name).is_running() or restart
 
         if self.server_cert.enabled and not self.server_cert.available:
             # We have a TLS relation in place, but the cert isn't there
             logger.info("Loki stopped")
             self._loki_container.stop(self._service_name)
 
-        elif restart:
+        # Now that we for sure have a layer, we can check if the service is running
+        elif not self._loki_container.get_service(self._service_name).is_running() or restart:
             self._loki_container.restart(self._name)
-            logger.info("Loki (re)started")
+            logger.info("Loki started")
```

Comment:
Imho it would help with debugging if we added here _why_ loki is being (re)started. is it because the service was not running, or because we just made a change to the config?